cli_tools/common/utils/compute/node_affinity_label_parser.go

```diff
@@ -0,0 +1,66 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License
+
+package computeutils
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/api/compute/v1"
+)
+
+// ParseNodeAffinityLabels parses sole tenant affinities
+// labels - array of strings with node affinity label info. Each label is of the following format:
+// <key>,<operator>,<value>,<value2>... where <operator> can be one of: IN, NOT.
+// For example: workload,IN,prod,test is a label with key 'workload' and values 'prod' and 'test'
+func ParseNodeAffinityLabels(labels []string) ([]*compute.SchedulingNodeAffinity, error) {
+	var nodeAffinities []*compute.SchedulingNodeAffinity
+
+	if labels == nil || len(labels) == 0 {
```

Comment:
This is a bit redundant, just have len(labels) == 0 or remove this check completly as you can range over a nil slice

Reply:
Done. Also added more tests to cover empty/nil labels flags